types/react-router/index.d.ts

```diff
@@ -146,6 +146,56 @@ export function matchPath<Params extends { [K in keyof Params]?: string }>(
     parent?: match<Params> | null,
 ): match<Params> | null;
 
+// Must only use abcdefghijklmnopqrstuvwxyz0123456789-_|
+type IsExtractableRegex<T extends string, L = Lowercase<T>> =
```

Comment:
Since we can't use version 4.5, I can't think of a cleaner way to check if it's alphanumeric. 

Previously I did 
```
type ExtractCharacters<T extends string> = T extends `${infer Left}${infer Right}`
    ? Left | ExtractCharacters<Right>
    : T;

type OnlyIncludes<T extends string, Allowed extends string> = ExtractCharacters<T> extends ExtractCharacters<Allowed>
    ? true
    : false;
```

But that gives a "too deep" error in version 4.4 and below